reconcile/gabi_authorized_users.py

```diff
@@ -0,0 +1,98 @@
+from datetime import datetime, date
+import logging
+import sys
+from typing import Dict, List, Optional
+
+import reconcile.queries as queries
+import reconcile.openshift_base as ob
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.openshift_resource import (OpenshiftResource as OR,
+                                                ResourceInventory as RI)
+from reconcile.utils.defer import defer
+
+QONTRACT_INTEGRATION = 'gabi-authorized-users'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+EXPIRATION_MAX = 90
+
+
+def construct_gabi_oc_resource(name: str, users: List[str]) -> OR:
+    body = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {
+            "name": name,
+            "annotations": {
+                "qontract.recycle": "true"
+            }
+        },
+        "data": {
+            "authorized-users.yaml": '\n'.join([u for u in users])
+        }
+    }
+    return OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION,
+              error_details=name)
+
+
+def fetch_desired_state(gabi_instances: List[Dict], ri: RI) -> None:
+    for g in gabi_instances:
+        exp_date = datetime.strptime(g['expirationDate'], '%Y-%m-%d').date()
+        if exp_date < date.today():
+            continue
+        elif (exp_date - date.today()).days > EXPIRATION_MAX:
+            logging.error(
+                f'The maximum expiration date of {g["name"]} '
+                f'shall not exceed {EXPIRATION_MAX} days form today')
+            sys.exit(1)
```

Comment:
Mostly just for my knowledge, is there a strong reason to use `sys.exit` versus `RunnerException`?  I see the former much more in the code, but the latter will exit 1 while logging the message in the exception via `cli.run_integration`.  Using `RunnerException` seems like it'd be easier to test, particularly if we care about the error message.